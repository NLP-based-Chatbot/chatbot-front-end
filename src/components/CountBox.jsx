import { Box, Card, Typography, makeStyles, Button } from '@material-ui/core'
import React from 'react'
import clsx from 'clsx';

const useStyles = makeStyles(theme => ({
  box: {
    margin: theme.spacing(3),
    height: "250px",
    width: "250px",
    paddingTop: theme.spacing(3)
  },
  weighted: {
    fontWeight: "600"
  },
  row: {
    paddingTop: theme.spacing(2),
  }
}))

const CountBox = ({ title, subtitle, count, changeGraph, disableGraph = false }) => {
  const classes = useStyles()
  return (
    <Card className={classes.box}>
      <Typography data-testid='title' variant="subtitle1" className={clsx(classes.weighted, classes.row)}>
        {title}
      </Typography>
      <Typography data-testid='subtitle' variant="subtitle2" className={classes.weighted}>
        {subtitle}
      </Typography>
      <Typography data-testid='count' variant="h3" className={clsx(classes.weighted, classes.row)}>
        {count}
      </Typography>
      {!disableGraph && <Box className={classes.row}>
        <Button
          data-testid='grp-btn'
          variant="outlined"
          color="secondary"
          className={classes.weighted}
          onClick={() => changeGraph()}
        >
          View Graph
        </Button>
      </Box>}
    </Card>
  )
}

export default CountBox