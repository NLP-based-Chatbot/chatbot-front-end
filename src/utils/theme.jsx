import { createTheme } from '@material-ui/core';

const theme = createTheme({
  typography: {
    fontFamily: ["Poppins", "sans-serif"].join(","),
  },
  palette: {
    type: 'light',
    primary: {
      light: "#0292BE",
      main: "#3A637E",
      dark: "#1A2C38",
      contrastText: "#fff",
    },
    secondary: {
      light: "#D3EFF8",
      main: "#BBDB9B",
      dark: "#ABC4A1",
      contrastText: "#000",
    },
  },
});

theme.typography.h6 = {
  [theme.breakpoints.up('md')]: {
    fontSize: "1.5rem",
    fontWeight: 500
  },
  [theme.breakpoints.down('md')]: {
    fontSize: "1.2rem",
    fontWeight: 500
  },
  [theme.breakpoints.down('sm')]: {
    fontSize: "0.8rem",
    fontWeight: 500
  },
  [theme.breakpoints.down('xs')]: {
    fontSize: "0.5rem",
    fontWeight: 500
  },
}

export default theme;