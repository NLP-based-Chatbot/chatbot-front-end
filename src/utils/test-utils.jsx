import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core";
import theme from "./theme";
import { ParallaxProvider } from "react-scroll-parallax";
import { BrowserRouter } from "react-router-dom";
import "@babel/polyfill";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import AuthReducer from '../store/slices/auth';

const customRender = (
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { auth: AuthReducer }, preloadedState }),
    ...renderOptions
  } = {}
) => {
  const AllTheProviders = ({ children }) => {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <ParallaxProvider>
            <ThemeProvider theme={theme}>{children}</ThemeProvider>;
          </ParallaxProvider>
        </BrowserRouter>
      </Provider>
    );
  };
  return render(ui, { wrapper: AllTheProviders, ...renderOptions });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
