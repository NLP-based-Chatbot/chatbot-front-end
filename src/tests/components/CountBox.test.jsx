/**
 * @jest-environment jsdom
 */

import { render } from "../../utils/test-utils";
import { screen, waitFor } from "@testing-library/react";
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import CountBox from "../../components/CountBox";
import userEvent from "@testing-library/user-event";

test("initial render", async () => {
  render(<CountBox />);
  screen.debug();
});

test("Display values for title, subtitle and count", async () => {
  const ChangeGraph = jest.fn();
  render(<CountBox title="test-title" subtitle="test-subtitle" count='20' changeGraph={ChangeGraph} disableGraph={false} />);
  expect(screen.getByTestId("title")).toHaveTextContent("test-title");
  expect(screen.getByTestId("subtitle")).toHaveTextContent("test-subtitle");
  expect(screen.getByTestId("count")).toHaveTextContent("20");
  expect(screen.getByTestId("grp-btn")).toBeInTheDocument();
  userEvent.click(screen.getByTestId("grp-btn"));
  await waitFor(() => {
    expect(ChangeGraph).toHaveBeenCalled()
  })
  //screen.debug();
});

test("snapshot test", async () => {
    const tree = renderer.create(<CountBox />).toJSON();
    expect(tree).toMatchSnapshot();
    //screen.debug();
  });
