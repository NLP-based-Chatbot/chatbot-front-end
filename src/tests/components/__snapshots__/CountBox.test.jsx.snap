// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-box-9 MuiPaper-elevation1 MuiPaper-rounded"
>
  <h6
    className="MuiTypography-root makeStyles-weighted-10 makeStyles-row-11 MuiTypography-subtitle1"
    data-testid="title"
  />
  <h6
    className="MuiTypography-root makeStyles-weighted-10 MuiTypography-subtitle2"
    data-testid="subtitle"
  />
  <h3
    className="MuiTypography-root makeStyles-weighted-10 makeStyles-row-11 MuiTypography-h3"
    data-testid="count"
  />
  <div
    className="MuiBox-root MuiBox-root-12 makeStyles-row-11"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-weighted-10 MuiButton-outlinedSecondary"
      data-testid="grp-btn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        View Graph
      </span>
    </button>
  </div>
</div>
`;
